src/vector_stores/pgvector.ts

```diff
@@ -0,0 +1,98 @@
+import pgpromise from 'pg-promise';
+import { IInitOptions } from 'pg-promise';
+import { registerType, toSql } from 'pgvector/pg';
+import { getEnvOrThrow } from '../config';
+import type { VectorStore, VectorizedDocument, VectorQuery, VectorQueryResult } from '../types';
+
+const initOptions: IInitOptions = {
+  async connect(e) {
+    await registerType(e.client);
+  },
+};
+
+const pgp = pgpromise(initOptions);
+const db = pgp(getEnvOrThrow('PG_DSN'));
```

Comment:
getEnv should only be used from command line tooling. The interface for the rest of the library should be driven by explicit options rather than assumptions about the surrounding environment. For example, if someone is not using PG, this file may still be required, and this line would then throw an error since they wouldn't have set this value in their environment.

I would move this inside the class constructor and make `dsn` an option to the constructor.